### PR TITLE
Adding upgrade notes for v3 to v4

### LINKED ISSUE #436: Create docs for upgrading to v4
There's been a number of breaking changes #435 #431 introduced recently.

Before releasing we'll need something in the upgrading doc and release notes to help people understand the changes. In particular, changes to when regexes are applied will affect running workloads.

### PR DESCRIPTION
**Pull request by stefansedich**
This PR adds the upgrade notes for v3 to v4 release which contains breaking changes which need to be documented.

Closes #436

### CODE REVIEW

File: docs/UPGRADING.md [hunk 1]
```diff
@@ -1,5 +1,25 @@
...
+Kiam changed significantly between v3.X and v4.0. Breaking changes are:
+
+- The role policy is now applied after the role ARN has been resolved, this may cause compatibility issues with existing `iam.amazonaws.com/permitted` restrictions.
+- StatsD metrics have been removed.
+- The agent gRPC keepalve flags have been renamed.
+
+We would suggest upgrading in the following way:
+
+1. TODO
+2. If you still require StatsD metrics you may need to look at something like [veneur-prometheus](https://github.com/stripe/veneur/tree/master/cmd/veneur-prometheus) to scrape the /metrics endpoint and push them to StatsD.
```
Comment (stefansedich) [inline docs/UPGRADING.md:14]: @pingles I am not sure what to say here, this the only possible prom->statsd route that I have hear of, any other ideas?

**Comment by stefansedich** [comment]
@pingles made a rough first cut at the upgrade docs. Let me know what you think and where I should look at adding more info.

File: docs/UPGRADING.md (hunk 1 above)
Comment (pingles) [reply docs/UPGRADING.md:16]: I'm not aware of any others. @Joseph-Irving aware of anything worth calling out?

If not, I'm hopeful that it affects a tiny number of users. Thanks for putting in something.

**Comment by pingles** [comment]
Thanks. 

There's a few other PRs that will have changed flags (#437 #440 #329 #427). I don't know whether it's worth calling out specific renames/drops in a table to help guide people?

**Comment by stefansedich** [comment]
> Thanks.
> 
> There's a few other PRs that will have changed flags (#437 #440 #329 #427). I don't know whether it's worth calling out specific renames/drops in a table to help guide people?

I made an attempt to move to tables splitting between the breaking agent flags and the new server flags.

**Review by pingles — APPROVED** [review]

**Pull request merged**